File: 0622-design-circular-queue/0622-design-circular-queue.py
class MyCircularQueue:

    def __init__(self, k: int):
        self.size = k
        self.circleQueue = []

    def enQueue(self, value: int) -> bool:
        if(len(self.circleQueue)!=self.size):
            self.circleQueue.append(value)

            return True
        return False

    def deQueue(self) -> bool:
        if(len(self.circleQueue)!=0):
            self.circleQueue.pop(0)

            return True
        return False

    def Front(self) -> int:
        if(len(self.circleQueue)!=0):
            return self.circleQueue[0]
        return -1

    def Rear(self) -> int:
        if(len(self.circleQueue)!=0):
            return self.circleQueue[-1]
        return -1

    def isEmpty(self) -> bool:
        return len(self.circleQueue)==0

    def isFull(self) -> bool:
        return len(self.circleQueue)==self.size


# Your MyCircularQueue object will be instantiated and called as such:
# obj = MyCircularQueue(k)
# param_1 = obj.enQueue(value)
# param_2 = obj.deQueue()
# param_3 = obj.Front()
# param_4 = obj.Rear()
# param_5 = obj.isEmpty()
# param_6 = obj.isFull()

File: 0622-design-circular-queue/0622-design-circular-queue.java
class MyCircularQueue {

    int size;
    ArrayList<Integer> li = new ArrayList<>();

    public MyCircularQueue(int k) {
       this.size = k;
    }
    
    public boolean enQueue(int value) {
        if(li.size()<size){
            li.add(value);
            return true;
        }
        return false;
    }
    
    public boolean deQueue() {
        if(li.size()!=0){
            li.remove(0);
            return true;
        }
        return false;
    }
    
    public int Front() {
        if(li.size()!=0){
            return li.get(0);
        }
        return -1;
    }
    
    public int Rear() {
        if(li.size()!=0){
            return li.get(li.size()-1);
        }
        return -1;
    }
    
    public boolean isEmpty() {
        return li.size()==0;
    }
    
    public boolean isFull() {
        return li.size()==size;
    }
}

/**
 * Your MyCircularQueue object will be instantiated and called as such:
 * MyCircularQueue obj = new MyCircularQueue(k);
 * boolean param_1 = obj.enQueue(value);
 * boolean param_2 = obj.deQueue();
 * int param_3 = obj.Front();
 * int param_4 = obj.Rear();
 * boolean param_5 = obj.isEmpty();
 * boolean param_6 = obj.isFull();
 */